src/gsy_e/models/strategy/external_strategies/__init__.py

```diff
@@ -344,13 +349,13 @@ def _offer_aggregator_impl(
 
             replace_existing = arguments.pop("replace_existing", True)
 
-            assert self.can_offer_be_posted(
-                arguments["energy"],
-                arguments["price"],
-                available_energy,
-                market,
-                time_slot=time_slot,
-                replace_existing=replace_existing)
+            if not self.can_offer_be_posted(arguments["energy"],
+                                            arguments["price"],
+                                            available_energy,
+                                            market,
+                                            time_slot=time_slot,
+                                            replace_existing=replace_existing):
+                raise OrderCanNotBePosted()
```

Comment:
Nit: I'd recommend using the other indentation style (called hanging indentation, as [mentioned in PEP 8](https://www.python.org/dev/peps/pep-0008/#indentation)). I originally thought it was just a personal preference, but I found out that it has some objective advantages:
1. it reduces the amount of empty white space, so it's less likely to break the max length of a line;
2. it avoids flake8 errors. If we rename `can_offer_be_posted` to something else with more/less characters, flake8 will raise errors regarding the indentation of all the arguments. This wouldn't happen using hanging indents.

Reply:
Also, I think that you don't need to instantiate the exception. This would work:
```python
raise OrderCanNotBePosted
```